src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -348,29 +413,141 @@ public sealed class FileInfoFormatting
             /// <summary>
             /// Gets or sets the style for directories.
             /// </summary>
-            public string Directory { get; set; } = "\x1b[44;1m";
+            public string Directory
+            {
+                get => _directory;
+                set => _directory = ValidateNoContent(value);
+            }
+
+            private string _directory = "\x1b[44;1m";
 
             /// <summary>
             /// Gets or sets the style for symbolic links.
             /// </summary>
-            public string SymbolicLink { get; set; } = "\x1b[36;1m";
+            public string SymbolicLink
+            {
+                get => _symbolicLink;
+                set => _symbolicLink = ValidateNoContent(value);
+            }
+
+            private string _symbolicLink = "\x1b[36;1m";
 
             /// <summary>
             /// Gets or sets the style for executables.
             /// </summary>
-            public string Executable { get; set; } = "\x1b[32;1m";
+            public string Executable
+            {
+                get => _executable;
+                set => _executable = ValidateNoContent(value);
+            }
+
+            private string _executable = "\x1b[32;1m";
+
+            /// <summary>
+            /// Custom dictionary handling validation of extension and content.
+            /// </summary>
+            public sealed class FileExtensionDictionary : IEnumerable
```

Comment:
It's not clear why this implements `IEnumerable`, but if it does, it may as well implement `IEnumerable<KeyValuePair<string, string>>`

Reply:
I'm removing this, I thought formatting was relying on it, but it doesn't